frontend/app/src/App.tsx

```diff
@@ -1061,6 +1061,11 @@ export class App extends PureComponent<Props, State> {
         newDialog,
         sessionEvent.scriptCompilationException?.message ?? "No message"
       )
+    } else if (sessionEvent.type === "scriptChangedOnDisk") {
+      // If runOnSave is enabled, automatically trigger a rerun with fresh context info
+      if (this.state.userSettings.runOnSave) {
+        this.widgetMgr.sendUpdateWidgetsMessage(undefined, true)
```

Comment:
The auto rerun flag is only meant for reruns triggered by the fragment auto-rerun mechanism. For a full rerun, just doing `sendUpdateWidgetsMessage(undefined)` is fine.